drivers/overlay/overlay.go

```diff
@@ -162,3 +150,46 @@ func (d *driver) configure() error {
 func (d *driver) Type() string {
 	return networkType
 }
+
+func (d *driver) nodeJoin(node net.IP, self bool) {
+	d.Lock()
+	if self {
+		d.bindAddress = node.String()
+	} else {
+		d.neighIP = node.String()
+	}
+	d.Unlock()
+	if d.bindAddress != "" && !d.isSerfAlive() {
+		err := d.serfInit()
+		if err != nil {
+			logrus.Errorf("initializing serf instance failed: %v", err)
+			return
+		}
+	}
+
+	if d.serfInstance != nil && d.neighIP != "" {
+		err := d.serfJoin()
```

Comment:
We are trying to call Join this node multiple times into the cluster. I don't know the full implication of it. So it's better to keep the old behavior and just join once.